design/ingress-status-loadbalancer.md

```diff
@@ -0,0 +1,101 @@
+# Ingress Status Loadbalancer Support
+
+Status: _Draft_
+
+This proposal describes how to add support for the Ingress.Status.LoadBalancer field to Contour.
+
+## Goals
+
+- Populate the status.loadbalancer field in all networking.k8s.io/v1beta.Ingress objects managed by Contour.
+- Contour pass the Ingress compliance tests for networking.k8s.io/v1beta.Ingress objects.
+
+## Non Goals
+
+- Similar functionality for IngressRoute (deprecated) and HTTPProxy are out of scope.
+
+## Background
+
+The Kubernetes Ingress object provides a field on the object's status to record address details for how access the ingresses virtualhosts from outside the cluster.
+This is known as the status loadbalancer information after the name of the fields on the Ingress Object.
+
+Details from the Kubernetes spec [are here](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#loadbalancerstatus-v1-core).
+
+## High-Level Design
+
+There are two components to this solution
+
+1. Discoverying the loadbalancer information
+2. Writing the loadbalacner information to the ingress objects.
+
+The former, the discovery will need to support 
+
+1. Hard coded IP/address details (actually a list, and it can be an address or an ip)
```

Comment:
Nit: 'it can be a domain name or an ip address' is probably slightly more accurate.